src/components/UserActivationPage/index.jsx

```diff
@@ -16,50 +17,39 @@ const USER_ACCOUNT_POLLING_TIMEOUT = 5000;
 
 const UserActivationPage = ({
   match,
-  authentication,
-  userAccount,
-  fetchUserAccount,
 }) => {
+  const user = getAuthenticatedUser();
+  const {
+    username, roles, isActive,
+  } = user;
   const [isPollingUserAccount, setIsPollingUserAccount] = useState(false);
   const enterpriseSlug = useMemo(
     () => match.params.enterpriseSlug,
     [match.params],
   );
   const { addToast } = useContext(ToastsContext);
 
-  useEffect(
-    () => {
-      let timeout;
-      if (authentication?.username && userAccount?.loaded && !userAccount?.isActive) {
-        // user is authenticated and we finished hydrating the full user metadata, but
-        // the user has not verified their email address. we start polling for their user
-        // metadata every USER_ACCOUNT_POLLING_TIMEOUT milliseconds.
-        setIsPollingUserAccount(true);
-        timeout = setTimeout(() => {
-          fetchUserAccount(authentication.username);
-        }, USER_ACCOUNT_POLLING_TIMEOUT);
-      }
-      return () => {
-        if (timeout) {
-          clearInterval(timeout);
-        }
-      };
-    },
-    [authentication, userAccount],
-  );
+  useInterval(() => {
+    if (username && !isActive) {
+      setIsPollingUserAccount(true);
+      hydrateAuthenticatedUser();
+    } else if (isActive) {
+      setIsPollingUserAccount(false);
+    }
+  }, USER_ACCOUNT_POLLING_TIMEOUT);
 
-  if (authentication?.username) {
+  if (username) {
     // user is authenticated, but doesn't have any JWT roles so redirect the user to
     // `:enterpriseSlug/admin/register` to display the proper error message.
-    if (!authentication.roles?.length) {
+    if (!roles?.length) {
       return (
         <Redirect to={`/${enterpriseSlug}/admin/register`} />
       );
     }
 
     // user data is hydrated with a verified email address, so redirect the user
     // to the default page in the Admin Portal.
-    if (userAccount?.loaded && userAccount?.isActive) {
+    if (isActive !== undefined && isActive) {
```

Comment:
to follow the pattern in place already should this be defined as a var too? `isActive !== undefined && isActive` for readability I guess could be called `isUserLoadedAndActive` ?

Reply:
I don't feel strongly about it. Do you?

Reply:
nah, given only one usage here not really, so yeah for this PR let's let go of it. I had initially thought I saw more than one usage of it, but since no, it does not really matter here...